### PR TITLE
update to new format-specific redaction types

### PR DESCRIPTION
**Pull request by Hamms**
will require comparable updates to https://github.com/code-dot-org/remark-plugins and https://github.com/code-dot-org/redactable-markdown

### CODE REVIEW

**Comment by bethanyaconnor** [comment]
Is it possible to move/add tests to this repo?

**Comment by Hamms** [comment]
Note that this repo does have its own tests.

It would probably be possible to move some of our other tests into this repo, but I don't think it's actually desired. This repo is intended to provide specifically those plugins needed to enable redaction and restoration, not to define the plugins themselves (that's what https://github.com/code-dot-org/remark-plugins is for). Most of the tests we want to do are on the plugins themselves, and those tests are provided by the two repositories that rely on this one.

File: src/redact.js [hunk 1]
```diff
@@ -40,36 +40,40 @@
...
-      let value = "";
-
-      var self = this;
-      var exit = self.enterLink();
-      if (node.children) {
-        value = self.all(node).join('');
+    visitors.inlineRedaction = function(node) {
+      let exit;
+      if (node.redactionType === "link" || node.redactionType === "image") {
+        exit = this.enterLink();
```
Comment (bethanyaconnor) [inline src/redact.js]: Why are link and image special cased?

Comment (Hamms) [reply src/redact.js]: So they don't escape the urls; see https://github.com/remarkjs/remark/blob/049b4033aaace271e4559604262b302d84b6234c/packages/remark-stringify/lib/compiler.js#L20-L23

**Pull request merged**